cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -331,13 +334,31 @@ instance ( IsShelleyBasedEra era
          , ToJSON (Core.TxOut ledgerera)
          , Share (Core.TxOut (ShelleyLedgerEra era)) ~ Interns (Shelley.Credential 'Shelley.Staking (Ledger.Crypto (ShelleyLedgerEra era)))
          ) => ToJSON (DebugLedgerState era) where
-  toJSON (DebugLedgerState newEpochS) = object [ "lastEpoch" .= Shelley.nesEL newEpochS
-                                          , "blocksBefore" .= Shelley.nesBprev newEpochS
-                                          , "blocksCurrent" .= Shelley.nesBcur newEpochS
-                                          , "stateBefore" .= Shelley.nesEs newEpochS
-                                          , "possibleRewardUpdate" .= Shelley.nesRu newEpochS
-                                          , "stakeDistrib" .= Shelley.nesPd newEpochS
-                                          ]
+  toJSON (DebugLedgerState newEpochS) =
+    object
+    [ "lastEpoch" .= Shelley.nesEL newEpochS
+    , "blocksBefore" .= Shelley.nesBprev newEpochS
+    , "blocksCurrent" .= Shelley.nesBcur newEpochS
+    , "stateBefore" .= Shelley.nesEs newEpochS
+    , "possibleRewardUpdate" .= Shelley.nesRu newEpochS
+    , "stakeDistrib" .= Shelley.nesPd newEpochS
+    ]
+  toEncoding (DebugLedgerState newEpochS) =
+    let !nesEL     = Shelley.nesEL newEpochS
+        !nesBprev  = Shelley.nesBprev newEpochS
+        !nesBcur   = Shelley.nesBcur newEpochS
+        !nesEs     = Shelley.nesEs newEpochS
+        !nesRu     = Shelley.nesRu newEpochS
+        !nesPd     = Shelley.nesPd newEpochS
+    in Aeson.pairs $ mconcat
+    [ "lastEpoch"             .= nesEL
+    , "blocksBefore"          .= nesBprev
+    , "blocksCurrent"         .= nesBcur
+    , "stateBefore"           .= nesEs
+    , "possibleRewardUpdate"  .= nesRu
+    , "stakeDistrib"          .= nesPd
+    ]
```

Comment:
NewEpochState is fully strict in all its fields, so there are no reasons to force them individually,
also I ment to say it on you previous PRs we should try avoiding duplication:
```suggestion
    $ nesPairs newEpochS
  toEncoding (DebugLedgerState newEpochS) = Aeson.pairs $ mconcat $ nesPairs newEpochS

nesPairs !newEpochS =
    [ "lastEpoch" .= Shelley.nesEL newEpochS
    , "blocksBefore" .= Shelley.nesBprev newEpochS
    , "blocksCurrent" .= Shelley.nesBcur newEpochS
    , "stateBefore" .= Shelley.nesEs newEpochS
    , "possibleRewardUpdate" .= Shelley.nesRu newEpochS
    , "stakeDistrib" .= Shelley.nesPd newEpochS
    ]
```